dlt/common/configuration/specs/aws_credentials_pure.py

```diff
@@ -0,0 +1,30 @@
+from typing import Optional, TYPE_CHECKING, Dict
+
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.specs import CredentialsConfiguration, CredentialsWithDefault, configspec
+from dlt import version
+
+
+@configspec
+class AwsCredentialsPure(CredentialsConfiguration, CredentialsWithDefault):
```

Comment:
yeah that should be like that from the very start. I'd just keep the convention we have in GcpCredentials
1. you can keep it in one file
2. AwsCredentialsPure -> rename to AwsCredentialsWithoutDefaults
3. AwsCredentialsPure -> do not inherit from CredentialsWithDefault
4. AwsCredentials -> inherit from CredentialsWithDefault

there are several places (ie. dbt helper) which checks for CredentialsWithDefault inheritance and then calls the CredentialsWithDefault interface. so it should be implemented where we really do that (check and retrieve default credentials)